src/libraries/Common/src/System/IO/FileSystem.Attributes.Windows.cs

```diff
@@ -112,6 +112,19 @@ internal static int FillAttributeInfo(string? path, ref Interop.Kernel32.WIN32_F
             return errorCode;
         }
 
+        internal static int FillAttributeInfo(SafeFileHandle fileHandle, ref Interop.Kernel32.WIN32_FILE_ATTRIBUTE_DATA data)
+        {
+            if (!Interop.Kernel32.GetFileInformationByHandle(
+                fileHandle,
+                out Interop.Kernel32.BY_HANDLE_FILE_INFORMATION fileInformationData))
+            {
+                return Marshal.GetLastWin32Error();
```

Comment:
It looks like both call sites are throwing an exception in response to an error here.  Why return the error code and force all call site to check it rather than just throwing the exception from here?

Reply:
Because `System.IO.FileSystem.AccessControl` also includes this file, we have no access to the internal property `SafeFileHandle.Path` - So throwing must be outside of this file.

```
E:\external\dotnet\runtime\src\libraries\Common\src\System\IO\FileSystem.Attributes.Windows.cs(122,92): error CS1061: "
SafeFileHandle" enthält keine Definition für "Path", und es konnte keine zugängliche Path-Erweiterungsmethode gefunden
werden, die ein erstes Argument vom Typ "SafeFileHandle" akzeptiert (möglicherweise fehlt eine using-Direktive oder ein
 Assemblyverweis). [E:\external\dotnet\runtime\src\libraries\System.IO.FileSystem.AccessControl\src\System.IO.FileSyste
m.AccessControl.csproj]
```

Reply:
Why does this function need to be in this file? It's obviously not used by that other project as the method is new.

Reply:
@stephentoub SafeFileHandle.Path is internal and System.IO.FileSystem.AccessControl does not have access to this internal property.

This file seems to be consumed by CoreLib & S.IO.AccessControl (see build failure)

Reply:
I understand that.  I'm asking why does this new function you added need to be added to this file rather than just putting it as a private helper in FileSystem.Windows.cs in corelib where it's consumed?

Reply:
see one of my last commits